### PR TITLE
Pentatonictritones/dw 139/create end time for ibal

### PR DESCRIPTION
**Pull request by pentatonictritones**
1. Two superfluid resolvers for:
a. Liquidation protection
b. Allowing an endDate / duration to be put on streams

2. Deployment details and OZ scripts

3. Small cleanup

### CODE REVIEW

**Comment by 0xtuytuy** [comment]
@pentatonictritones @zayk00v where are we here ?

**Comment by pentatonictritones** [comment]
> @pentatonictritones @zayk00v where are we here ?

@zayk00v reviewed it and we deployed it together. It is on both mumbai and polygon.

Maybe @zayk00v wants to do some formatting / cleanup before this is merged with master?

**Review by alexanderem49 — APPROVED** [review]
LGTM

File: contracts/Farming/Polygon/IbAlluo.sol [hunk 1]
```diff
@@ -374,48 +377,81 @@ contract IbAlluo is
...
+        cfaV1Lib.updateFlowByOperator(_msgSender(), receiver,  ISuperfluidToken(superToken),flowRate);
+        emit UpdatedFlow(_msgSender(), receiver, flowRate);
+    }
+
+    // /// @notice Wraps and creates flow with context
+    // /// @dev Forces transfer of ibAlluo to the StIbAlluo contract then mints StIbAlluos to circumvent having to sign multiple transactions to create streams
+    // /// @param receiver The recipient of the streamed flow
+    // /// @param flowRate The amount of ibAlluos per second to be streamed (decimals 10**18)
+    // /// @param toWrap The amount of ibAlluos to automatically wrap (recommend wrapping entire ibALluo balance initially)
+    // function createFlow(address receiver, int96 flowRate, uint256 toWrap, bytes memory ctx, bytes memory userData, uint256 timestamp) external {
```
Comment (0xtuytuy) [inline contracts/Farming/Polygon/IbAlluo.sol:421]: do we need to keep this as a huge comment?

File: hardhat.config.ts [hunk 2]
```diff
@@ -40,32 +40,64 @@ const config: HardhatUserConfig = {
     polygon: {
       url: process.env.POLYGON_URL,
       gasPrice: "auto",
+<<<<<<< HEAD
```
Comment (0xtuytuy) [inline hardhat.config.ts:43]: looks like there are some merge headers left here

File: scripts/deploy/priceFeeds/deployFeedsPhase1.ts [hunk 3]
```diff
@@ -88,18 +88,18 @@ async function main() {
...
-            id: 2,
-            oracle: "0x73366fe0aa0ded304479862808e02506fe556a98"
-        },
-        {
-            // https://data.chain.link/polygon/mainnet/fiat/gbp-usd
-            name: "GBP",
-            id: 3,
-            oracle: "0x099a2540848573e94fb1ca0fa420b00acbbc845a"
-        }
+        // {
```
Comment (0xtuytuy) [inline scripts/deploy/priceFeeds/deployFeedsPhase1.ts:91]: do we need to keep this comment ?

File: test/farming/StIbAlluo.ts [hunk 4]
```diff
@@ -191,17 +190,43 @@ describe("IbAlluoUSD and Handler", function () {
...
+            let permissions = await realIbAlluo.connect(signers[1]).formatPermissions();
+            let userData = "0x"
+
+            let operationData = ethers.utils.defaultAbiCoder.encode(["bytes", "bytes"],[permissions,userData])
+
+            let operation = {operationType: 201, target: cfaAddress, data: operationData}
+            let operationArray = [operation];
+            console.log(operationArray);
+
+            // https://dashboard.tenderly.co/pentatonictritones/project/simulator/fd708bad-b2d9-4e7b-8949-93a07834fdd7
```
Comment (0xtuytuy) [inline test/farming/StIbAlluo.ts:219]: Do we want to keep this again ?

**Review by 0xtuytuy — CHANGES_REQUESTED** [review]
A few things to look at.

File: contracts/Farming/Polygon/IbAlluo.sol (hunk 1 above)
Comment (Jorge-Lopes) [reply contracts/Farming/Polygon/IbAlluo.sol:421]: Comment deleted on commit [b44c08d](https://github.com/GetAlluo/liquidity-direction-protocol/commit/b44c08d8ce2487eb8aa39ee5efb029e97aaf03fe)

File: hardhat.config.ts (hunk 2 above)
Comment (Jorge-Lopes) [reply hardhat.config.ts:43]: Conflict resolved on commit [b44c08d](https://github.com/GetAlluo/liquidity-direction-protocol/commit/b44c08d8ce2487eb8aa39ee5efb029e97aaf03fe)

File: scripts/deploy/priceFeeds/deployFeedsPhase1.ts (hunk 3 above)
Comment (Jorge-Lopes) [reply scripts/deploy/priceFeeds/deployFeedsPhase1.ts:91]: Comment deleted on commit [b44c08d](https://github.com/GetAlluo/liquidity-direction-protocol/commit/b44c08d8ce2487eb8aa39ee5efb029e97aaf03fe)

**Review by 0xtuytuy — APPROVED** [review]
lgtm

**Pull request merged**